### PR TITLE
LPS-129404 fix tests

### PR DESCRIPTION
**Pull request by marcelabc**

### CODE REVIEW

**Comment by rodrigopaulino** [comment]
Just started reviewing.

**Review by rodrigopaulino — CHANGES_REQUESTED** [review]
@marcelabc, it is not recommended to reference services in order to add data during upgrade ([here](https://github.com/liferay-data-engine/liferay-portal/pull/350/files#diff-ecce0ede8890ad524549589a5a0517f1e2bdc5748038ba85694ab15663599714R207-R209) and [here](https://github.com/liferay-data-engine/liferay-portal/pull/350/files#diff-ecce0ede8890ad524549589a5a0517f1e2bdc5748038ba85694ab15663599714R232-R235)). We should use prepared statements instead to avoid possible concurrency issues.

**Comment by ealonso** [comment]
Hey @marcelabc @rodrigopaulino,

Using Services from other modules is allowed, it is not allowed to use services from the same module. If you have a dependency between two modules, then the module is going to wait for the other one to execute the upgrade process.

cc: @achaparro

**Comment by rodrigopaulino** [comment]
> Hey @marcelabc @rodrigopaulino,
> 
> Using Services from other modules is allowed, it is not allowed to use services from the same module. If you have a dependency between two modules, then the module is going to wait for the other one to execute the upgrade process.
> 
> cc: @achaparro

Hi, @ealonso. I see. Thanks for explaining that to me. Do you know what happens if, eventually, we end up introducing a reference in that service to the dynamic-data-mapping-service module? Would that cause a dependency loop issue?

**Comment by ealonso** [comment]
In this case you are not going to include a dependency between services modules you are going to add a dependency with the api module, so it should not be a problem.

**Comment by rodrigopaulino** [comment]
I understand. Thanks, @ealonso.

**Comment by rodrigopaulino** [comment]
Hi, @marcelabc. Why aren't we using the new convertDDMFormDataDefinition in the [DDMStructureVersion upgrade](https://github.com/liferay-data-engine/liferay-portal/pull/350/files#diff-323af2a62da3a6a16b6a6fbe663c7ebeb3c5b2046c02f51f3f2b00ccb400790eR220-R223) as well?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMDataDefinitionConverterImpl.java [hunk 1]
```diff
@@ -94,6 +102,25 @@ public String convertDDMFormDataDefinition(
 		return DDMFormSerializeUtil.serialize(ddmForm, _ddmFormSerializer);
 	}
 
+	@Override
+	public String convertDDMFormDataDefinition(
+			String dataDefinition, long groupId, long parentStructureId,
+			long parentStructureLayoutId, long structureId)
+		throws Exception {
+
+		DDMForm ddmForm = DDMFormDeserializeUtil.deserialize(
```
Comment (rodrigopaulino) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMDataDefinitionConverterImpl.java:111]: Can we call the deserialize directly in the convertDDMFormDataDefinition call?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMDataDefinitionConverterImpl.java [hunk 2]
```diff
@@ -158,7 +186,59 @@ public String convertDDMFormLayoutDataDefinition(
...
+			long dataDefinitionId, long dataLayoutId, DDMForm ddmForm,
+			List<String> fieldNames, long groupId)
+		throws Exception {
+
+		Map<String, DDMFormField> ddmFormFieldsMap =
+			ddmForm.getDDMFormFieldsMap(true);
+
+		for (String fieldName : fieldNames) {
+			_deDataDefinitionFieldLinkLocalService.addDEDataDefinitionFieldLink(
+				groupId, _portal.getClassNameId(DDMStructureLayout.class),
```
Comment (rodrigopaulino) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMDataDefinitionConverterImpl.java:208]: Apparently, getClassNameId has some complex logic that can be avoided for the second time if we assign its result to a variable.

**Comment by rodrigopaulino** [comment]
Hi, @marcelabc. Can you reorder the commits so the "add dependencies" one comes before its usage?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMDataDefinitionConverterImpl.java (hunk 1 above)
Comment (marcelabc) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMDataDefinitionConverterImpl.java:111]: This class already has _ddmFormDeserializer as its reference. If I deserialize directly in the call, I would have to repeat the same reference in the other class. Can we let it this way?

**Comment by marcelabc** [comment]
ci:test:sf

**Comment by marcelabc** [comment]
ci:test:relevant

**Comment by ealonso** [comment]
ci:test:relevant

**Comment by marcelabc** [comment]
ci:test:relevant

**Comment by marcelabc** [comment]
ci:forward

**Pull request closed without merging**